src/lib/ssr-api.ts

```diff
@@ -1,22 +1,45 @@
 import { getResourceCollection } from 'next-drupal'
 import { NODE_TYPES } from '@/lib/drupalApiTypes'
-import { Tags } from '@/lib/types'
+import { EventsQueryParams } from '@/lib/types'
 import { baseEventQueryParams, baseArticlePageQueryParams, baseTprUnitQueryParams } from './params'
 
-export const getEvents = async ({ tags }: Tags) => {
-  const eventParams = () =>
+export const getEvents = async (queryParams: EventsQueryParams) => {
+  const { tags, locationId } = queryParams
+
+  let eventParams = () =>
     baseEventQueryParams()
       .addSort('field_end_time', 'ASC')
       .addPageLimit(3)
 
-    if (tags) {
-      const filteredEventParams = () =>
-        eventParams()
-          .addFilter('field_tags', tags, 'IN')
-          .getQueryObject()
+  if (tags && locationId) {
+    const filteredEventParams = () =>
+      eventParams()
+        .addFilter('field_tags', tags, 'IN')
+        // @ts-ignore
+        .addFilter('field_location_id', locationId)
```

Comment:
If you change `locationId?: number | null` => `locationId?: string | null` and change `locationId` type to string where it is used and also change `drupal_internal__id` type to string in types.ts you should be able to remove these ts-ignores.